ring/lifecycler_test.go

```diff
@@ -596,7 +595,7 @@ func TestTokensOnDisk(t *testing.T) {
 	require.NoError(t, services.StartAndAwaitRunning(context.Background(), r))
 	defer services.StopAndAwaitTerminated(context.Background(), r) //nolint:errcheck
 
-	tokenDir, err := ioutil.TempDir(os.TempDir(), "tokens_on_disk")
+	tokenDir, err := os.MkdirTemp("", "tokens_on_disk")
```

Comment:
can we use `t.TempDir()`? No need to do manual deletion via `defer` then.

Reply:
@pstibrany good point, didn't think of it myself :)